ci_framework/roles/env_op_images/tasks/main.yml

```
@@ -0,0 +1,105 @@
+    PATH: "{{ cifmw_path }}"
+  block:
+    - name: Get images from the csv
+      ansible.builtin.shell: |
+        csv_name=$(oc get csv -n {{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }} -l operators.coreos.com/openstack-operator.openstack-operators -o=jsonpath='{.items[*].metadata.name}')
+        oc get csv -n {{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }} ${csv_name} -o=jsonpath='{.spec.install.spec.deployments[0].spec.template.spec.containers[1].env[*]}'
+      register: csv_images
+
+    - name: Extract env variable name and images
+      loop: "{{ csv_images.stdout| regex_findall('\\{[^}]+\\}') }}"
+      ansible.builtin.set_fact:
+        csv_image_list: "{{ csv_image_list | default([]) + [item | from_json] }}"
+
+    - name: Add operator images to the dictionary
+      loop: "{{ csv_image_list }}"
+      ansible.builtin.set_fact:
+        cifmw_openstack_service_images_content: "{{ cifmw_openstack_service_images_content | default({}) | combine({item.name: item.value}) }}"
+
+    - name: Get all the pods in openstack-operator namespace
+      kubernetes.core.k8s_info:
+        kind: Pod
+        namespace: "{{ cifmw_install_yamls_defaults['OPERATOR_NAMESPACE'] }}"
+        kubeconfig: "{{ cifmw_openshift_kubeconfig }}"
+        api_key: "{{ cifmw_openshift_token | default(omit)}}"
+        context: "{{ cifmw_openshift_context | default(omit)}}"
+        field_selectors:
+          - status.phase=Running
+      register: pod_list
+
+    - name: Retrieve openstack-operator-index pod
+      vars:
+        selected_pod: "{{ pod_list.resources| selectattr('metadata.generateName', 'equalto', 'openstack-operator-index-') | list|first}}"
+      ansible.builtin.set_fact:
+        cifmw_install_yamls_vars_content:
+          OPENSTACK_IMG: "{{ selected_pod.status.containerStatuses[0].imageID }}"
+
+    - name: Get operator images and pods
+      vars:
+        selected_pod: "{{ pod_list.resources | selectattr('metadata.generateName', 'contains', 'rabbitmq-cluster-operator-') | list | first }}"
+      ansible.builtin.set_fact:
+        cifmw_openstack_operator_images_content:
+          RABBITMQ_OP_IMG: "{{ selected_pod.status.containerStatuses[0].imageID }}"
+        selected_pods: "{{ pod_list.resources | rejectattr('metadata.generateName', 'contains', 'openstack-operator-index-') | rejectattr('metadata.generateName', 'contains', 'rabbitmq-cluster-operator-') }}"
+
+    - name: Add operator images to the dictionary
+      loop: "{{ selected_pods }}"
+      ansible.builtin.set_fact:
+        cifmw_openstack_operator_images_content: "{{ cifmw_openstack_operator_images_content | combine({ item.metadata.labels['openstack.org/operator-name'] | upper ~ '_OP_IMG': item.status.containerStatuses[1].imageID}) }}"
+
+    - name: Write images to file
+      ansible.builtin.template:
+        src: templates/operator_images.yml.j2
+        dest: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file }}"
```

Instead of using template, we can use ansible.builtin.copy module here like this
```suggestion
    - name: Write images to file
      ansible.builtin.copy:
        content:  |
             ----
             cifmw_install_yamls_vars:
                  {{ cifmw_install_yamls_vars_content | to_nice_yaml | replace('\n', '\n  ')}}
             cifmw_openstack_operator_images:
                   {{ cifmw_openstack_operator_images_content | to_nice_yaml | replace('\n', '\n  ')}}
              cifmw_edpm_prepare_extra_vars:
                  {{ cifmw_openstack_service_images_content | to_nice_yaml | replace('\n', '\n  ')}}
         dest: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file }}"
```

---

In order to replace `\n`, we can also use `trim` filters.

---

Or even better:
```yaml

- name: Write images to file
  vars:
    _content:
      cifmw_install_yamls_vars: "{{ cifmw_install_yamls_vars_content }}"
      cifmw_openstack_operator_images: "{{ cifmw_openstack_operator_images_content }}"
      cifmw_edpm_prepare_extra_vars: "{{ cifmw_openstack_service_images_content }}"
  ansible.builtin.copy:
    dest: "{{ cifmw_env_op_images_dir }}/artifacts/{{ cifmw_env_op_images_file }}"
    content: "{{ _content | to_nice_yaml }}"
```

---

Anyway, as I said in the review call, is really this the expected output for a role that gathers container images? A set of dicts grouped by who knows what criteria? The last thing I'd expect here is seeing variables like `cifmw_install_yamls_vars` or `cifmw_edpm_prepare_extra_vars` as keys of the dict. I know that this format is convenient as it will be easy to consume, but common, let's try to make roles decoupled from each other.